src/Controls/tests/TestCases.HostApp/Issues/Issue28023.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System.Collections.ObjectModel;
+using System.Windows.Input;
+
+namespace Maui.Controls.Sample.Issues;
+[XamlCompilation(XamlCompilationOptions.Compile)]
+[Issue(IssueTracker.Github, 28023, "I2 Entering the vertical and horizontal lists again Monkeys still retain the spacing value changed last time", PlatformAffected.All)]
+public partial  class Issue28023: NavigationPage
+{
+	public Issue28023() : base(new _28023Page())
+	{
+
+	}
+}
+
+public class _28023Page : ContentPage
+{
+	public _28023Page()
+	{
+		var button = new Button
+		{
+			AutomationId = "NavigatedPageButton",
+			Text = "GoTo Main Page"
+		};
+
+		button.Clicked += (sender, e) =>
+		{
+			Navigation.PushAsync(new _28023MainPage());
+		};
+
+		Content = new StackLayout
+		{
+			Children =
+			{
+				button
+			}
+		};
+	}
+}
+
+public class _28023MainPage : ContentPage
+{
+	CollectionView collectionView;
+
+	public _28023MainPage()
+	{
+		BindingContext = new _28023ViewModel();
+
+		var updateButton = new Button
+		{
+			AutomationId = "MainPageButton",
+			Text = "Update Item Spacing",
+			Margin=5,
+			HorizontalOptions = LayoutOptions.Center,
+			VerticalOptions = LayoutOptions.Center
+		};
+		updateButton.Clicked += ButtonClicked;
+
+		var navigationButton = new Button
+		{
+			AutomationId = "PopButton",
+			Text = "GoTo Previous page",
+			Margin=5,
+			HorizontalOptions = LayoutOptions.Center,
+			VerticalOptions = LayoutOptions.Center
+		};
+		navigationButton.Clicked += NavigationButtonClicked;
+
+		collectionView = new CollectionView
+		{
+			ItemTemplate = new DataTemplate(() =>
+			{
+				var image = new Image { HeightRequest = 100 };
+				image.SetBinding(Image.SourceProperty, "Source");
+
+				var label = new Label
+				{
+					HorizontalTextAlignment = TextAlignment.Center
+				};
+				label.SetBinding(Label.TextProperty, "Text");
+
+				return new StackLayout
+				{
+					Orientation = StackOrientation.Vertical,
+					Spacing = 10,
+					BackgroundColor = Colors.Beige,
+					Padding = 10,
+					Children = { image, label }
+				};
+			})
+		};
+		collectionView.SetBinding(ItemsView.ItemsSourceProperty, "Items");
+
+		var stackLayout = new StackLayout
+		{
+			Children =
+			{
+				updateButton,
+				navigationButton,
+				collectionView
+			}
+		};
+
+		Content = stackLayout;
+	}
+
+	void ButtonClicked(object sender, EventArgs e)
+	{
+		if (collectionView?.ItemsLayout is LinearItemsLayout linearItemsLayout)
+		{
+			linearItemsLayout.ItemSpacing = 100;
```

Comment:
Just to confirm, are the test snapshots using a 100 ItemSpacing?

Reply:
@jsuarezruiz, The test snapshots are not using a 100 value for ItemSpacing. In the current implementation, when navigating back to the page via Navigation.PopAsync(), the ItemSpacing applied to the CollectionView is cleared. This behavior is consistent across all platforms.